src/routes/posts.js

```diff
@@ -20,8 +20,16 @@ const router = express.Router()
 // GET ALL ARTICLES
 
 const fetchPosts = catchAsync( async(req, res) => {
+	let feed
+	const {isFlagged} = req.query
 
-	const feed = await postService.fetchPosts()
+	
+	if (isFlagged === 'true' && req.user.role === 'admin'){
+		feed = await postService.fetchFlaggedPosts()
+	}
+	 else {
+		feed = await postService.fetchPosts()
+	 }
```

Comment:
the `fetchPosts` should take in the parameter and handle the query, the only difference between them is a `WHERE` clause

we will be likely be having more parameters so its mostly about forming the right query based on the parameters than create a totally different function

Reply:
i am not sure how to do this